core/shared/platform/common/posix/posix_memmap.c

```diff
@@ -235,6 +240,23 @@ os_munmap(void *addr, size_t size)
     }
 }
 
+#if WASM_HAVE_MREMAP != 0
+void *
+os_mremap(void *old_addr, size_t old_size, size_t new_size)
+{
+    void *ptr = mremap(old_addr, old_size, new_size, MREMAP_MAYMOVE);
+
+    if (ptr == MAP_FAILED) {
```

Comment:
When mremap fails, had better mmap a new memory with new_size, copy old memory data to it and then munmap the old memory.